src/nbl/asset/utils/ISPIRVOptimizer.cpp

```diff
@@ -46,8 +46,9 @@ nbl::core::smart_refctd_ptr<ICPUBuffer> ISPIRVOptimizer::optimize(const uint32_t
         &spvtools::CreateIfConversionPass,
         &spvtools::CreateStripDebugInfoPass,
         &spvtools::CreateTrimCapabilitiesPass,
-        &spvtools::CreateAggressiveDCEPass
+        &spvtools::CreateAggressiveDCEPass,
         &spvtools::CreateRemoveUnusedInterfaceVariablesPass,
+        &spvtools::CreateEliminateDeadInputComponentsSafePass,
     };
```

Comment:
I want you to have a think if there'a less fragile way to initialize this array (maybe via a consteval function) so its not susceptible to getting the ordering wrong between the nable enum and SPV tools optimizer pass token. Maybe something like a list of
```cpp
array[NABLA_ENUM] = &Token;
```